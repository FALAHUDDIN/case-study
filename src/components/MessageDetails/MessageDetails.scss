.MessageDetails {
  background: #ccc;

  &__message {
    background: #fff;
  }
}