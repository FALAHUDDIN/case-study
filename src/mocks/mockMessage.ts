import {IMessage} from "../types/IMessage";

export const MockMessage: IMessage = {
    id: 1,
    isRead: false,
    avatar: "person1.svg",
    sender: "Falah",
    message: "testmessage",
    received: "01/01/2020",
    subject: "testsubject",
    isArchived: false
}
export const MockMessageRead: IMessage = {
    ...MockMessage,
    isRead: true
}
export const MockMessageArchived: IMessage = {
    ...MockMessage,
    isArchived: true
}