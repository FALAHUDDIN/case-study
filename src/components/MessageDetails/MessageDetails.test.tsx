import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import {MessageDetails, Props} from "./MessageDetails";
import {MockMessage} from "../../mocks/mockMessage";

describe('<MessageDetails/>', () => {
    let wrapper: ShallowWrapper;
    let props: Props = {
        message: undefined,
        markRead: jest.fn(),
        markUnread: jest.fn(),
        archiveMessage: jest.fn(),
    };

    describe('without message', () => {
        beforeEach(() => {
            wrapper = shallow(<MessageDetails {...props} />)
        })

        it('displays info text', () => {
            expect(wrapper.find('.MessageDetails__noMessageSelected').length).toBe(1)
            expect(wrapper.find('.MessageDetails__noMessageSelected').text()).toBe('Please select a message')
        })
    })

    describe('with message', () => {
        beforeEach(() => {
            const mergedProps = {...props, message: MockMessage}
            wrapper = shallow(<MessageDetails {...mergedProps} />)
        })

        it('match snapshot', () => {
            expect(wrapper).toMatchSnapshot()
        })

        it('displays subject', () => {
            expect(wrapper.find('.MessageDetails__subject').text()).toEqual(MockMessage.subject)
        })

        it('displays message', () => {
            expect(wrapper.find('.MessageDetails__message').text()).toEqual(MockMessage.message)
        })

        it('displays received date', () => {
            expect(wrapper.find('.MessageDetails__received').text()).toContain(MockMessage.received)
        })

        it('clicking unread button calls markUnread callback', () => {
            wrapper.find('.MessageDetails__unreadButton').simulate('click')
            expect(props.markUnread).toBeCalled()
        })

        it('clicking archive button calls archiveMessage callback', () => {
            wrapper.find('.MessageDetails__archiveButton').simulate('click')
            expect(props.archiveMessage).toBeCalled()
        })
    })
});