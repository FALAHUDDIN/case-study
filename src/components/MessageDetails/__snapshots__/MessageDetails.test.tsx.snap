// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageDetails/> with message match snapshot 1`] = `
<Styled(MuiBox)
  className="MessageDetails"
  p={2}
>
  <Styled(MuiBox)
    p={1}
  >
    <strong
      className="MessageDetails__subject"
    >
      testsubject
    </strong>
     
    <span
      className="MessageDetails__received"
    >
      (
      01/01/2020
      )
    </span>
  </Styled(MuiBox)>
  <Styled(MuiBox)
    className="MessageDetails__message"
    p={1}
  >
    testmessage
  </Styled(MuiBox)>
  <Styled(MuiBox)
    p={1}
  >
    <WithStyles(ForwardRef(Button))
      className="MessageDetails__unreadButton"
      color="primary"
      onClick={[Function]}
      size="small"
      variant="contained"
    >
      Mark as unread
    </WithStyles(ForwardRef(Button))>
     Â 
    <WithStyles(ForwardRef(Button))
      className="MessageDetails__archiveButton"
      color="secondary"
      onClick={[Function]}
      size="small"
      variant="contained"
    >
      Archive
    </WithStyles(ForwardRef(Button))>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
